daniel/pass/ps.c

```diff
@@ -0,0 +1,94 @@
+#include "ps.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <dirent.h>
+
+#define EXE_PATH_STRING "/proc/%d/exe"
+#define STAT_PATH_STRING "/proc/%d/stat"
+#define CMD_PATH_STRING "/proc/%d/cmdline"
+int_array_t getProcessIdList() {
+    DIR *directory;
+    int_array_t pidList;
+    int counter = 0;
+    struct dirent *file;
+    directory = opendir("/proc");
+    if (directory) {
+        file = readdir(directory);
+        while (file != NULL) {
+            if (atoi(file->d_name)) {
+                pidList.data[counter++] = atoi(file->d_name);
+            }
+        file = readdir(directory);
+        }
+        pidList.length = counter;
+        closedir(directory);
+    }
+    return pidList;
+}
+
+void replace(char *string, int length, char old, char new) {
+    for (int counter = 0; counter < length; counter++) {
+        if (string[counter] == old) {
+            string[counter] = new;
+        }
+    }
+    string[length - 1] = 0;
+}
+
+int getProcessName(process_t *process) {
+    char statFileName[MAX_PROC_PATH];
+    sprintf(statFileName, STAT_PATH_STRING, process->pid);
+    FILE *statFile = fopen(statFileName, "r");
+    return fscanf(statFile, "%*s %s", process->name);
+}
+
+
+void getProcessCommand(process_t *process) {
+    char cmdFileName[MAX_PROC_PATH];
+    char *cmdData = (char *) calloc(MAX_COMMAND, sizeof(char));
+    int copiedLength = 0;
+    if (!cmdData) {
+        fprintf(stderr, "error allocating memory.");
```

Comment:
You checked for successful allocation but you didn't exit the program or anything, so you'll have errors when you try to read/free it